### PR TITLE
Add Matter Node Device

### PR DESCRIPTION
**Pull request by balloob**
Some Matter nodes are a bridge that expose multiple devices. This adds support for this.

See architecture drawing for latest architecture that includes Matter Node Device: https://docs.google.com/drawings/d/1hxhOXKquTUooWH6sSfCfw02fyfIOYL9MoxJf6EBUQ9o/edit

### CODE REVIEW

File: matter_server/client/model/mappable_device.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+def mappable_devices_from_node(node: MatterNode) -> list[MappableDevice]:
+    """Split a node in mappable devices."""
+    if not node.bridge_device:
+        return [MappableNodeDevice(node)]
+
+    devices = []
+
+    for device in node.devices:
+        if device.device_type == device_types.BridgedDevice:
```
Comment (agners) [inline matter_server/client/model/mappable_device.py:25]: Shouldn't that be `!=`?

Comment (agners) [reply matter_server/client/model/mappable_device.py:25]: Hm, from what I can tell we exclude the `BridgedDevice` anyways when building `devices` list not?

**Review by agners — COMMENTED** [review]
I guess there might be other cases where we need to split a node into multiple devices, but I guess that would need another `MappableDevice`. Needs to be implemented when it comes up.

File: matter_server/client/model/mappable_device.py (hunk 1 above)
Comment (balloob) [reply matter_server/client/model/mappable_device.py:25]: we exclude `Bridge`, not `BridgedDevice` . We use `BridgedDevice` to indicate which endpoints are their own devices.

**Comment by balloob** [comment]
Marking as draft because I first want to merge entity description <> device mapping and then pick a better name for this too.

**Comment by balloob** [comment]
Verified works.

File: tests/matter_experimental/test_mapping.py [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+    async def connect(self):
+        self.server_info = Mock(compressedFabricId=MOCK_COMPR_FABRIC_ID)
+
+    async def listen(self, driver_ready):
+        self.driver = Driver(self)
+        driver_ready.set()
+        self.mock_client_disconnect = asyncio.Event()
+        await self.mock_client_disconnect.wait()
+
+    # async def async_send_command(
```
[the remark this answers is not in the document]
Comment (balloob) [reply tests/matter_experimental/test_mapping.py:47]: Removed 4f60fd92468d97fd79e7ed9809be08c223875938

**Pull request merged**